### PR TITLE
(5.x) Merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
OpenCV Contrib: https://github.com/opencv/opencv_contrib/pull/3721
OpenCV Extra: None

#25107 from catree:fix_mistake_reprojectImageTo3D_doc
#25306 from utibenkei:fix_build_of_dynamic_framework_for_visionos
#25343 from asmorkalov:as/HAL_bilaterialFilter
#25351 from Kumataro:fix25073_format_g
#25365 from MaximSmolskiy:take-first-suitable-neighbor-in-ChessBoardDetector-findQuadNeighbors
#25377 from eplankin:3.4
#25378 from AleksandrPanov:move_charuco_tutorial
#25383 from asmorkalov:as/win32_arm_ci
#25385 from asmorkalov:as/new_contours_cleanup
#25387 from fengyuentau:complete-float16_t-renaming
#25388 from asmorkalov:as/jpeg-turbo-diagnostics
#25391 from Kumataro:refix25073
#25395 from susumu-iino:fix-dnn-plugin-build-win32
#25396 from mshabunin:test-tag-contours
#25405 from VadimLevin:dev/vlevin/ios-framework-privacy-info
#25406 from VadimLevin:dev/vlevin/numpy-array-typing-constraints
#25407 from Kumataro:fix25401
#25413 from Linaname:#25404

Previous "Merge 4.x": #25384

### CODE REVIEW

**Comment by opencv-alalek** [comment]
From 4.x changes:

```.diff
+  Windows10-ARM64:
+    uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-W10-ARM64.yaml@main
+
```

Is there anything for 5.x?

File: doc/tutorials/objdetect/table_of_content_objdetect.markdown [hunk 1]
```diff
@@ -3,4 +3,8 @@ Object Detection (objdetect module) {#tutorial_table_of_content_objdetect}
 
 -   @subpage tutorial_aruco_detection
 -   @subpage tutorial_aruco_board_detection
+-   @subpage tutorial_charuco_detection
+-   @subpage tutorial_charuco_diamond_detection
+-   @subpage tutorial_aruco_calibration
+-   @subpage tutorial_aruco_faq
 -   @subpage tutorial_barcode_detect_and_decode
```
Comment (opencv-alalek) [inline doc/tutorials/objdetect/table_of_content_objdetect.markdown:10]: broken prev/next links.

File: doc/tutorials/objdetect/aruco_faq/aruco_faq.markdown [hunk 2]
```diff
@@ -0,0 +1,191 @@
+Aruco module FAQ {#tutorial_aruco_faq}
+================
+
+@next_tutorial{tutorial_barcode_detect_and_decode}
+@prev_tutorial{tutorial_aruco_calibration}
```
[the remark this answers is not in the document]
Comment (asmorkalov) [reply doc/tutorials/objdetect/aruco_faq/aruco_faq.markdown:4]: Fixed.

File: doc/tutorials/objdetect/table_of_content_objdetect.markdown (hunk 1 above)
Comment (asmorkalov) [reply doc/tutorials/objdetect/table_of_content_objdetect.markdown:10]: Fixed.

**Comment by asmorkalov** [comment]
Windows ARM CI for 5.x is in progress: https://github.com/opencv/ci-gha-workflow/pull/170

File: modules/core/include/opencv2/core/cvdef.h [hunk 3]
```diff
@@ -933,7 +933,9 @@ inline hfloat hfloatFromBits(ushort w) {
 #endif
 }
 
+#if !defined(__OPENCV_BUILD) && !(defined __STDCPP_FLOAT16_T__) && !(defined __ARM_NEON)
 typedef hfloat float16_t;
+#endif
```
Comment (fengyuentau) [inline modules/core/include/opencv2/core/cvdef.h:938]: This should be dropped in 5.x. That's to say no float16_t in 5.x even for external user.

**Comment by shijunz** [comment]
> From 4.x changes:
> 
> ```diff
> +  Windows10-ARM64:
> +    uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-W10-ARM64.yaml@main
> +
> ```
> 
> Is there anything for 5.x?

o , which branch? it is ok now?

**Pull request merged**